File: programmers/Hash/p42576.java
package programmers.Hash;

import java.util.HashMap;

public class p42576 {
    public static void main(String[] args) {
        String[] participant = {"marina","josipa","nikola","vinko","filipa"};
        String[] completion = {"josipa","filipa","marina","nikola"};
        String result = solution(participant, completion);
        System.out.println(result);
    }
    public static String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> map = new HashMap<>();

        for(String person : completion){
            map.put(person, map.getOrDefault(person, 0) + 1);
        }
        for(String person : participant){
            map.put(person, map.getOrDefault(person, 0) - 1);
            if(map.get(person) < 0)
                return person;
        }
        return answer;
    }
}

File: programmers/Hash/p42576.js
const participant = ['mislav','stanko','mislav','ana'];
const completion = ['stanko','mislav','ana'];
console.log(solution(participant, completion));

function solution(participant, completion) {
    var newList = completion.reduce(function(acc, c){
        acc[c] = acc[c] ? (acc[c]+1) : 1;
        return acc;
    },{});
    return participant.find(function(c){
        if(newList[c]) {
            newList[c] -= 1;
        } else {
            return true;
        }
    });
}